pkg/shared/accountmgmtutil/ams.go

```diff
@@ -36,20 +36,22 @@ func CheckTermsAccepted(ctx context.Context, spec *remote.AmsConfig, conn connec
 
 // QuotaSpec - contains quota name and remaining quota count
 type QuotaSpec struct {
-	Name         string
-	Quota        int
-	BillingModel string
+	Name          string
+	Quota         int
+	BillingModel  string
+	CloudAccounts *[]amsclient.CloudAccount
 }
 
-func GetUserSupportedInstanceType(ctx context.Context, spec *remote.AmsConfig, conn connection.Connection) (quota *QuotaSpec, err error) {
-	userInstanceTypes, err := GetUserSupportedInstanceTypes(ctx, spec, conn)
-	if err != nil {
-		return nil, err
-	}
-
-	amsType := PickInstanceType(userInstanceTypes)
+type MarketplaceInfo struct {
+	BillingModel   string
+	Provider       string
+	CloudAccountID string
+}
 
-	return amsType, nil
+type OrgQuotas struct {
```

Comment:
While this is good I would myself keep all Quota spec as array and use filter flags (billingModel) to filter them out. 

There is big decision that we can take:

- Have trial as quota object (it will be always there) thus if logic is `len(quotas) == 1 { return quotas[0] } 
- Ignore trial and do not return quota  for it (it will be more accurate from business logic point of view , but it will require rewrite of the create.go logic.